app/views/notes/index.html.erb

```diff
@@ -8,8 +9,37 @@
 
 <% if @notes.empty? %>
   <h4><%= t ".no_notes" %></h4>
-
 <% else %>
+  <%= form_tag(url_for("controller" => "notes", "action" => "index"), "method" => :get, "data" => { "turbo" => true, "turbo-frame" => "pagination", "turbo-action" => "advance" }) do %>
+    <div class="row gx-2">
+      <div class="col-md-auto mb-3">
+        <%= label_tag :status, t(".status") %>
+        <%= select_tag  :status,
+                        options_for_select([[t(".all"), "all"], [t(".open"), "open"], [t(".closed"), "closed"]], params[:status]),
+                        :include_blank => t(".select_status"),
+                        :class => "form-select" %>
+      </div>
+      <div class="col-md-auto mb-3">
+        <%= label_tag :note_type, t(".note_type") %>
```

Comment:
I wouldn't call this a "note type". It's an interaction type between the user and the note.

And then there's a question if this kind of filtering can be done efficiently because here you have `INNER JOIN "note_comments"` twice.

Reply:
You're right. I've renamed that into interaction type which is a better naming for this.